### ISSUE TITLE
Wrong disassembling of x64 code

### ISSUE THREAD

**Issue by rfalke** [uxmal/reko#507]
Subject: https://github.com/rfalke/decompiler-subjects/blob/master/from_rec/hd2/x64_elf_from_rec_gcc4O/subject.exe
Method: dumpline

Just the first opcode (bytes 0x41 0x57) at address 0x4006c0 is disassembled by repo as "push rdi" while https://defuse.ca/online-x86-assembler.htm and https://alexaltea.github.io/capstone.js/ says "push   r15".

There are also other problems like at 0x4006eb the assignment of 0x10 to esp.

Tested with 0.7.2.

**Comment by uxmal** [comment]
Thanks Raimar. It's embarrassing that this bug has been in the disassembler since the very beginning. I have  a fix available shortly. Can you build reko yourself or do you need a drop?

**Comment by rfalke** [comment]
I looked at the reko output when I started my first x64 program. My disassembler and the one from reko had the same bug. It took me quite some time to find this out. Afterwards I started verifying the new disassembler with big random files with the output of udis86 (https://github.com/vmt/udis86). This gives confidence. Also with the VEX stuff.

I do not plan to build reko myself. A new version would be nice but it is not pressing.
